src/lib/holocene/accordion.svelte

```diff
@@ -2,65 +2,79 @@
   import Icon from '$holocene/icon/icon.svelte';
   import Badge from '$holocene/badge.svelte';
   import type { IconName } from './icon/paths';
+  import type { HoloceneComponentProps } from 'src/types/holocene';
+
+  interface $$Props extends HoloceneComponentProps<'div'> {
+    title: string;
+    id: string;
+    subtitle?: string;
+    icon?: IconName;
+    open?: boolean;
+    disabled?: boolean;
+    readOnly?: boolean;
+    error?: string;
+  }
+
   export let title: string;
-  export let subtitle: string = '';
-  export let icon: IconName = null;
-  export let open: boolean = false;
-  export let disabled: boolean = false;
-  export let readOnly: boolean = false;
-  export let error: string = '';
+  export let id: string;
+  export let subtitle = '';
+  export let icon = null;
+  export let open = false;
+  export let disabled = false;
+  export let readOnly = false;
+  export let error = '';
+
+  let className = '';
+  export { className as class };
 
   $: open = disabled ? true : open;
+
+  const toggleAccordion = () => {
+    if (disabled || readOnly) return;
+    open = !open;
+  };
 </script>
 
-<section
-  class="flex w-full cursor-default flex-row rounded-lg border border-gray-300 bg-white p-8 text-primary {$$props.class}"
+<div
+  class="flex w-full cursor-default flex-col rounded-lg border border-gray-300 bg-white p-8 text-primary {className}"
+  {...$$restProps}
 >
-  <div class="w-full">
-    <div
-      class="accordion-open flex {!readOnly ? 'cursor-pointer' : ''} flex-col"
-      class:open
-      class:disabled
-      on:click={() => {
-        if (disabled || readOnly) return;
-        open = !open;
-      }}
-    >
-      <div class="space-between flex flex-row">
-        <h2 class="flex w-full items-center gap-2 text-lg font-medium">
-          {#if icon}<Icon name={icon} />{/if}
-          {title}
-        </h2>
-        <div class="mr-1" on:click|stopPropagation>
-          <slot name="action" />
-        </div>
-        {#if !readOnly}
-          <Icon
-            name={open ? 'chevron-up' : 'chevron-down'}
-            class={disabled ? 'text-gray-500' : 'text-primary'}
-          />
-        {/if}
+  <button
+    id="{id}-trigger"
+    aria-expanded={open}
+    aria-controls="{id}-content"
```

Comment:
We could use `title` here instead, but there is the risk that it won't be unique. We might want to think about creating a generic util to generate a unique id for use cases such as this.

```suggestion
    aria-controls="{title}-content"
```

Another approach could be to rework this to be a `<li>` inside of a `<ul>` and use the index + title for the id.

Ultimately, I think `aria-expanded` is the most important addition here 😅